### PR TITLE
improvement on documentation

### PR DESCRIPTION
**Pull request by sturkmen72**
related issue : #9520 
i want to add some more link in [examples page](http://docs.opencv.org/master/examples.html)

for a trial purpose i added related images top of code page. @mshabunin  i am not sure if it is a good idea. what is your opinion?

you can see changes on documentation
[contours2.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d0/d2a/contours2_8cpp-example.html)
[facedetect.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/db/d3a/facedetect_8cpp-example.html)
[houghlines.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d2/d7b/houghlines_8cpp-example.html)
[lsd_lines.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d8/dd4/lsd_lines_8cpp-example.html)
[grabcut.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/de/dd0/grabcut_8cpp-example.html)
[morphology2.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d8/dc0/morphology2_8cpp-example.html)
[Morphology_1.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/dc/dfa/Morphology_1_8cpp-example.html)
[Smoothing.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d4/dbd/Smoothing_8cpp-example.html)
[Sobel_Demo.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d4/dba/Sobel_Demo_8cpp-example.html)

### CODE REVIEW

**Comment by sturkmen72** [comment]
@mshabunin is there a way to put a link of [tutorial_sobel_derivatives](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d2/d2c/tutorial_sobel_derivatives.html) in [Sobel_Demo.cpp](http://pullrequest.opencv.org/buildbot/export/pr/9526/docs/d4/dba/Sobel_Demo_8cpp-example.html)

**Comment by mshabunin** [comment]
@sturkmen72 , looks like you should add it the `@example` block, the same way as you add an image.
```
/** @example Sobel_Demo.cpp
Sample code using Sobel and/or Scharr OpenCV functions to make a simple Edge Detector
![Sample screenshot](Sobel_Derivatives_Tutorial_Result.jpg)
See @ref tutorial_sobel_derivatives for more details.
 */
```
You can also use `@ref <link> "<text>"` form:
```
Check @ref tutorial_sobel_derivatives "the corresponding tutorial" for more details
```

File: samples/cpp/edge.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ static void onTrackbar(int, void*)
 
 static void help()
 {
-    printf("\nThis sample demonstrates Canny edge detection\n"
+    printf("\nThis sample demonstrates using Canny edge detection\n"
```
Comment (vpisarev) [inline samples/cpp/edge.cpp]: "using" is irrelevant here

File: samples/cpp/facedetect.cpp [hunk 2]
```diff
@@ -8,7 +8,7 @@ using namespace cv;
 
 static void help()
 {
-    cout << "\nThis program demonstrates the cascade recognizer. Now you can use Haar or LBP features.\n"
+    cout << "\nThis program demonstrates using the cv::CascadeClassifier class to detect objects (Face + eyes). You can use Haar or LBP features.\n"
```
Comment (vpisarev) [inline samples/cpp/facedetect.cpp]: using =&gt; the use of

File: samples/cpp/tutorial_code/ImgTrans/Sobel_Demo.cpp [hunk 3]
```diff
@@ -1,77 +1,117 @@
...
 int main( int argc, char** argv )
 {
+  cv::CommandLineParser parser(argc, argv,
+                               "{@input   |../data/lena.jpg|input image}"
+                               "{ksize   k|1|ksize (hit 'K' to increase its value)}"
+                               "{scale   s|1|scale (hit 'S' to increase its value)}"
+                               "{delta   d|0|delta (hit 'D' to increase its value)}"
+                               "{help    h|false|show help message}");
+
+  cout << "Sample code using Sobel or Scharr OpenCV functions to make a simple Edge Detector\n\n";
```
Comment (vpisarev) [inline samples/cpp/tutorial_code/ImgTrans/Sobel_Demo.cpp]: The sample uses Sobel or Scharr OpenCV functions for edge detection

**Comment by sturkmen72** [comment]
@vpisarev updated according to your review.

**Comment by vpisarev** [comment]
thanks! :+1:

**Pull request merged**